src/msg/async/Event.cc

```diff
@@ -425,6 +425,7 @@ int EventCenter::process_events(int timeout_microseconds,  ceph::timespan *worki
     numevents += process_time_events();
 
   if (external_num_events.load()) {
+    numevents += external_num_events.load();
```

Comment:
I don't think so, this is out of external_lock, it may change during the period

Reply:
Yes,. I'll move this under external_lock.

Reply:
no, it will unlcok and lock, so it's not safe to handle this here